apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/PatientLinkBuilder.java

```diff
@@ -0,0 +1,97 @@
+package gov.cms.bfd.server.war.stu3.providers;
+
+import ca.uhn.fhir.rest.api.Constants;
+import ca.uhn.fhir.rest.server.exceptions.InvalidRequestException;
+import java.util.List;
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Bundle.BundleEntryComponent;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/** A link builder for Patient resources using bene-id cursors */
+public final class PatientLinkBuilder implements LinkBuilder {
+  private final UriComponents components;
+  private final Integer count;
+  private final String cursor;
+
+  public static final String PARAM_CURSOR = "cursor";
+
+  public PatientLinkBuilder(String requestString) {
+    components = UriComponentsBuilder.fromUriString(requestString).build();
```

Comment:
I'm initializing with a string vs RequestDetail because it makes it easier to test the class.